test/spec_response.rb

```diff
@@ -419,20 +419,24 @@ def object_with_each.each
     header['content-length'].must_equal '9'
   end
 
-  it "correctly updates content-type when writing when initialized with body" do
-    obj = Object.new
-    def obj.each
-      yield 'foo'
-      yield 'bar'
-    end
-    ["foobar", ["foo", "bar"], obj].each do
-      r = Rack::Response.new(["foo", "bar"])
-      r.write('baz')
-      _, header, body = r.finish
-      str = "".dup; body.each { |part| str << part }
-      str.must_equal "foobarbaz"
-      header['content-length'].must_equal '9'
-    end
+  it "correctly updates content-length when writing when initialized with Array body" do
+    r = Rack::Response.new(["foo"])
+    r.write('bar')
+    r.write('baz')
+    _, header, body = r.finish
+    str = "".dup; body.each { |part| str << part }
+    str.must_equal "foobarbaz"
+    header['content-length'].must_equal '9'
+  end
+
+  it "correctly updates content-length when writing when initialized with String body" do
```

Comment:
🗒️ Now there are 3 specs, covering all three branching conditions of the `Rack::Response` initializer:

1. `nil` body
2. Array body
3. String body

The Array case is what is fixed by this PR.

Reply:
What about "Object that responds to `#each` body"?

Reply:
Good catch. I've added a test for that (see 578243c8) and then squashed to one commit.